src/drive/web/modules/public/SharingBanner.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState, useCallback } from 'react'
+
+import {
+  SharingBannerByLink,
+  SharingBannerCozyToCozy
+} from 'components/sharing/PublicBanner'
+import { useSharingInfos } from 'drive/web/modules/public/useSharingInfos'
+
+const BannerManager = () => {
+  const [isOpened, setIsOpened] = useState(true)
+  const onClose = useCallback(() => setIsOpened(false), [setIsOpened])
+
+  const {
+    loading,
+    discoveryLink,
+    sharing,
+    isSharingShortcutCreated
+  } = useSharingInfos()
```

Comment:
We execute two times this `useSharingInfos` hook. One in publicToolbar and one here. This hooks makes at least 3 queries. These queries can be huge, we should avoid to do that : 
- either by running only once the hook
- either by adding a cache layer to this useSharingInfo

Reply:
Suite à ce qu'on s'est dit, je n'ai pas réussi à mettre en place un context qui pourrait gérer du cache, les 2 composants étant sur le même parent et donc exécutés en même temps.
Je suis parti sur une solution plus simple : mettre le hook sur le parent et passer le résultat en prop aux composants en question (cf dernier commit)